dropshot_endpoint/src/endpoint.rs

```diff
@@ -1094,7 +1101,7 @@ mod tests {
                 /** handle "xyz" requests */
                 async fn handler_xyz(
                     _rqctx: MyRequestContext,
-                    query: Query<&'static str>,
+                    query: Query<QueryParams<'static>>,
                     path: Path<<X as Y>::Z>,
                 ) -> Result<HttpResponseOk<()>, HttpError> {
```

Comment:
this is a nit, but I'd love to get rid of all references to `HttpResponseOk<()>` as I **never** want people to do that. Find to defer that, but just saying it so one of us remembers.

Reply:
Good point, I'll try and change future ones at least.

Reply:
Updated all of the ones in #984 to use `HttpResponseUpdatedNoContent` btw.